tools/sync.js

```diff
@@ -44,153 +44,241 @@ var listenBlocks = function(config) {
       }
     }
   });
-  // Starts full sync when set to true in config
-  if (config.syncAll === true){
-    console.log('Starting Full Sync');
-    getOldestBlockDB(config);
-  }
-  // Starts full sync when set to true in config
-  if (config.patch === true){
-    console.log('Checking for missing blocks');
-    runPatcher(config);
-  }
 }
 /**
   If full sync is checked this function will start syncing the block chain from lastSynced param see README
 **/
-var syncChain = function(config,web3,nextBlock){
+var syncChain = function(config, nextBlock){
   if(web3.isConnected()) {
-    web3.eth.getBlock(nextBlock, true, function(error,blockData) {
-      if(error) {
-        console.log('Warning: error on getting block with hash/number: ' + nextBlock + ': ' + error);
-        getOldestBlockDB();
-      }else if(blockData == null) {
-        console.log('Warning: null block data received from the block with hash/number: ' + nextBlock);
-        getOldestBlockDB();
-      }else{
-        writeBlockToDB(config, blockData);
-        writeTransactionsToDB(config, blockData);
-      }
-    });
+    if (typeof nextBlock === 'undefined') {
+      prepareSync(config, function(error, startBlock) {
+        if(error) {
+          console.log('ERROR: error: ' + error);
+          return;
+        }
+        syncChain(config, startBlock);
+      });
+      return;
+    }
+
+    if( nextBlock == null ) {
+      console.log('nextBlock is null');
+      return;
+    } else if( nextBlock < config.startBlock ) {
```

Comment:
Sync will stop here and not proceed with newer blocks. Once sync hits 0 it should then start syncing later blocks.

Reply:
yes. Once `syncChain()` hits `0 block (or startBlock)`, it will do nothing.
but the `listenBlocks()` always listen the `latest` block to sync.